src/main/java/util/ApiQuotaManager.java

```diff
@@ -0,0 +1,169 @@
+package util;
+
+import backend.Logic;
+import backend.github.ApiQuotaInfo;
+import org.apache.logging.log4j.Logger;
+import ui.UI;
+import util.events.RefreshTimerTriggeredEvent;
+import util.events.RefreshTimerTriggeredEventHandler;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Manages the API quota usage and periodic refreshes of the data store.
+ */
+public class ApiQuotaManager {
+
+    private static final Logger logger = HTLog.get(ApiQuotaManager.class);
+
+    /**
+     * The amount that is set aside for manual operations by the user.
+     */
+    private static final int API_QUOTA_BUFFER = 200;
+
+    /**
+     * The default refresh period.
+     */
+    private static final int DEFAULT_REFRESH_PERIOD_IN_MINS = 1;
+
+    /**
+     * The previous amount of the available remaining API requests.
+     */
+    private int previousRemainingApiRequests = 0;
```

Comment:
Right, changed it. 

Hmm, setting it to 5000 is not good also. Another way is by calling the api once to get the initial value when the ApiQuotaManager instance is created. But that doesn't make much difference in the usefulness also. The algorithm needs at least one refresh iteration to get useful refresh rate, since it taps on the difference in api quota between refreshes.